src/System.Text.Primitives/System/Text/Encoding/SymbolTable.cs

```diff
@@ -107,6 +107,77 @@ public bool TryParse(ReadOnlySpan<byte> source, out Symbol symbol, out int bytes
 
         public abstract bool TryParse(ReadOnlySpan<byte> source, Span<byte> utf8, out int bytesConsumed, out int bytesWritten);
 
+        #region Public UTF-16 to UTF-8 helpers
+
+        public bool TryEncode(ReadOnlySpan<char> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten)
```

Comment:
There was a single extension method. It made the extension class overkill. This is a common enough function that we decided it made sense to put it directly on the class.